.github/workflows/build-and-test-Linux.yaml

```diff
@@ -0,0 +1,53 @@
+name: Build Linux partially with CPROVER tools
+
+on:
+  pull_request:
+    branches:
+      - '**'
```

Comment:
I'm a little hesitant about (another) CI job that runs on all PRs, specially one that we expect to report failures.

Reply:
Feel free to suggest another branch pattern. I can also target the develop and production branch only. That might result in late surprises.

Reply:
@TGWDB Why would you expect failures here?

Reply:
@nmanthey I'm no git expert and don't have a better suggestion. I agree that late failures are not a good idea.

@tautschnig  Based on the original PR text:

"This change adds a github workflow that compiles parts of the Linux kernel with goto-cc, and fails in case compilation fails. Furthermore, the input fails that lead to the failure are archived, to allow easy consumption and investigation."

I'm happy if this works without failures, but was (mis?)reading that we expect this to fail more often due to the target and perhaps related aspects of `cbmc`.

If we expect `goto-cc` to never fail this except for when we break something then I have no objection.